### PR TITLE
Refactored ProductMockDbContext

### PR DESCRIPTION
**Pull request by zinkoviktor**

### CODE REVIEW

File: NetCoreRestApi/DataLayer.EF/Configurations/ProductConfiguration.cs [hunk 1]
```diff
@@ -8,35 +8,24 @@ public class ProductConfiguration : IEntityTypeConfiguration<ProductEntity>
...
+                .Property(c => c.Name)
+                .HasMaxLength(50)
+                .IsRequired();
+            builder
+                .Property(p => p.Description)
+                .HasMaxLength(150);
+            builder
+                .HasMany(p => p.Categories)
+                .WithOne()
+                .HasForeignKey(c => c.Id);
```
Comment (mdemii) [inline NetCoreRestApi/DataLayer.EF/Configurations/ProductConfiguration.cs]: Please rework for Many to many

File: NetCoreRestApi/DataLayer.EF/Configurations/CategoryConfiguration.cs [hunk 2]
```diff
@@ -8,29 +8,23 @@ public class CategoryConfiguration : IEntityTypeConfiguration<CategoryEntity>
...
+                .Property(c => c.Name)
+                .HasMaxLength(50)
+                .IsRequired();
+            builder
+                .Property(c => c.Description)
+                .HasMaxLength(150);
+            builder
+                .HasOne<ProductEntity>()
+                .WithMany(p => p.Categories)
+                .HasForeignKey(c => c.Id);
```
Comment (mdemii) [inline NetCoreRestApi/DataLayer.EF/Configurations/CategoryConfiguration.cs]: Rework to Many to many

File: NetCoreRestApi/DataLayer.EF/ProductMockDbContext.cs [hunk 3]
```diff
@@ -1,78 +1,86 @@
...
+                .Entity<CategoryEntity>()
+                .HasData(_categoryEntities);
+            modelBuilder                
+                .Entity<ProductEntity>()                
+                .HasData(_productModels);            
+        }
+
+        private void SetupProducts()
+        {
+            _productModels = new List<ProductEntity>
```
Comment (mdemii) [inline NetCoreRestApi/DataLayer.EF/ProductMockDbContext.cs]: Specify here categories for each product entity

**Review by mdemii — CHANGES_REQUESTED** [review]
Needs to be reworked

File: NetCoreRestApi/DataLayer.EF/Configurations/ProductConfiguration.cs (hunk 1 above)
Comment (zinkoviktor) [reply NetCoreRestApi/DataLayer.EF/Configurations/ProductConfiguration.cs]: fixed

File: NetCoreRestApi/DataLayer.EF/Configurations/CategoryConfiguration.cs (hunk 2 above)
Comment (zinkoviktor) [reply NetCoreRestApi/DataLayer.EF/Configurations/CategoryConfiguration.cs]: fixed

File: NetCoreRestApi/DataLayer.EF/ProductMockDbContext.cs (hunk 3 above)
Comment (zinkoviktor) [reply NetCoreRestApi/DataLayer.EF/ProductMockDbContext.cs]: fixed

**Pull request merged**